lib/readyset/configuration.rb

```diff
@@ -2,38 +2,36 @@
 
 module Readyset
   class Configuration
-    attr_accessor :connection_url, :database_selector, :database_resolver,
-      :database_resolver_context
+    attr_accessor :database_url
 
     def initialize
-      @connection_url = ENV['READYSET_URL'] || default_connection_url
-      @database_selector = { delay: 2.seconds }
-      @database_resolver = Readyset::DefaultResolver
-      @database_resolver_context = nil
+      @database_url = ENV['READYSET_URL'] || default_connection_url
+    end
+
+    def self.current_config
+      configuration.inspect
     end
 
     def default_connection_url
       # Check if Rails is defined and if database configuration is available
-      if defined?(Rails) && Rails.application && Rails.application.config.database_configuration
-        # Fetch the environment-specific configuration
-        config = Rails.application.config.database_configuration[Rails.env]
+      'postgres://user:password@localhost:5432/readyset'
+    end
 
-        # Fetch the 'primary_replica' details if available, otherwise fallback to default database
-        replica_config = config['primary_replica'] || config
+    # @return [Readyset::Configuration] Readyset's current configuration
+    def self.configuration
+      @configuration ||= Configuration.new
```

Comment:
Hmm I think it might be a bit confusing to store an instance of `Configuration` as a class ivar in its own class -- were you trying to avoid storing `@configuration` in the `Readyset` module?

IMO, the scope of `Configuration` should just be a container for our config options that defines `attr_accessors` for each one, and then `Readyset` can define a `.configure` method that yields with `Readyset.configuration` -- what do you think?

Reply:
Huh, I'm looking at that now, trying to figure out why I did that. 

It might have been because I was looking into making a `Config` class to make `Configuration` more robust. Like limiting the assumptions of what `@instance_variables` are present in `Configuration`.  

But... I didn't actually do that lmao. `Configuration.configuration = Configuration.new` feels a bit silly now :rofl:.

Reply:
No worries, I do stuff like that all the time 😄